DESIGN.md

```diff
@@ -0,0 +1,200 @@
+# webcomponents.org v2 Design
+
+The webcomponents.org v2 relaunch is a from-scratch rebuild of the current webcomponents.org with up-to-date content, a more familiar and easier-to-maintain stack, and a simpler library-agnostic catalog system.
+
+## Stack
+
+The Stack is updated to be more familiar to modern JavaScript developers, including using a popular static site generator like 11ty, Node.js servers, and GraphQL.
+
+* Server
+  * Google Cloud Run
+  * Node 18+
+  * Google Firestore database
+  * GraphQL via graphql-helix and graphql-tools
+* Client
+  * Web components with Lit
+  * Apollo GraphQL Client
+  * Playground Elements
+* Static site generation
+  * eleventy
+* Schema
+  * GraphQL SDL
+  * graphql-codegen to generate TypeScript interfaces
+
+## Catalog
+
+The catalog is a standalone backend service with a GraphQL API that is accessed by the site's frontend server.
+
+### Custom Elements Manifest
+
+[Custom Elements Manifest](https://github.com/webcomponents/custom-elements-manifest) is a format for describing npm packages of custom elements, intended for applications like a catalog. Manifests are generated by various tools, like the [Custom Elements Manifest Analyzer](https://www.npmjs.com/package/@custom-elements-manifest/analyzer).
+
+The catalog will read custom elements manifests from npm. We detect manifests by reading the `"customElements"` field from `package.json` so that we don't have to download a package to see if it contains a manifest.
+
+### Indexing Packages
+
+#### Open-source, npm packages first
+
+The catalog will initially be oriented around open-source packages published to npm. This is where the vast majority of elements are now published.
+
+Cataloging publically available closed-source elements (those sold to the public for a fee) is not a goal, as that would require indexing proprietary information about where to retrieve elements, their license and cost, etc.
+
+Package sources other than npm, like GitHub packages or package-less distributions available via HTTP, or closed-source internal elements, may be included in the future. npm sources will be namespaced to prevserve the ability to add new sources.
+
+#### Permissionless importing
+
+We want to make importing packages as transparent as possible, so that they require no special permission or ownership of packages to import. Any anonymous users should be able to request that an npm package is imported. So we need to make importing fast, idempotent (when there is nothig to update), and rate-limited to protect against abuse and DOS.
```

Comment:
```suggestion
We want to make importing packages as transparent as possible, so that they require no special permission or ownership of packages to import. Any anonymous users should be able to request that an npm package is imported. So we need to make importing fast, idempotent (when there is nothing to update), and rate-limited to protect against abuse and DOS.
```